caddyconfig/caddyfile/lexer.go

```diff
@@ -356,6 +358,19 @@ func (t Token) NumLineBreaks() int {
 	return lineBreaks
 }
 
+// Clone returns a deep copy of the token.
+func (t Token) Clone() Token {
```

Comment:
I needed to make a `Clone()` because the token has some unexported fields that can't be copied from inside the `httpcaddyfile` package.